### PR TITLE
ARROW-12228: [CI] Create base image for conda environments

### PR DESCRIPTION
**Pull request by kszucs**
Using a base image we can reduce the number of installed packages for the conda-integration image. cc @pitrou

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -160,10 +161,31 @@ services:
...
+        arch: ${ARCH}
+        prefix: /opt/conda
+    volumes:
+      - .:/arrow:delegated
+
   conda-cpp:
     # C++ build in conda environment, including the doxygen docs.
     #
     # Usage:
+    #   docker-compose build conda
```
Comment (kszucs) [inline docker-compose.yml:188]: We should remove these instructions in favor of using `archery docker` command.

File: ci/docker/conda-integration.dockerfile [hunk 2]
```diff
@@ -27,8 +27,10 @@ ARG go=1.15
 
 COPY ci/conda_env_archery.yml /arrow/ci/
 RUN conda install -q \
+        --file arrow/ci/conda_env_cpp.yml \
         --file arrow/ci/conda_env_archery.yml \
         numpy \
+        compilers \
```
Comment (pitrou) [inline ci/docker/conda-integration.dockerfile:33]: If it's not useful for `conda-cpp`, is it useful here?

Comment (kszucs) [reply ci/docker/conda-integration.dockerfile:33]: It should be [in the conda-cpp](https://github.com/apache/arrow/pull/9905/files#diff-77e83f3866a7b845e723c3eb031ba36dcafc14d833ace16baecaae1a2056cf9eR29) image.

Comment (pitrou) [reply ci/docker/conda-integration.dockerfile:33]: Ah oops, sorry!

**Review by pitrou — APPROVED** [review]
+1 from me, this is a welcome improvement

**Comment by kou** [comment]
@github-actions crossbow submit -g conda

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**